flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/AutoScalerInfo.java

```diff
@@ -86,20 +87,24 @@ public AutoScalerInfo(Map<String, String> data) {
         configMap.setData(Preconditions.checkNotNull(data));
     }
 
-    @SneakyThrows
-    public SortedMap<Instant, Map<JobVertexID, Map<ScalingMetric, Double>>> getMetricHistory() {
+    public SortedMap<Instant, CollectedMetrics> getMetricHistory() {
         var historyYaml = configMap.getData().get(COLLECTED_METRICS_KEY);
         if (historyYaml == null) {
             return new TreeMap<>();
         }
 
-        return YAML_MAPPER.readValue(decompress(historyYaml), new TypeReference<>() {});
+        try {
+            return YAML_MAPPER.readValue(decompress(historyYaml), new TypeReference<>() {});
+        } catch (JsonProcessingException e) {
+            LOG.error(
+                    "Could not deserialize metric history, possibly the format changed. Discarding...");
+            return new TreeMap<>();
```

Comment:
This will clear all existing metrics when we deploy this new version.

Reply:
I think this will introduce a bug where only a single metric observation in a window will be considered: Fixed via https://issues.apache.org/jira/browse/FLINK-31866 or https://issues.apache.org/jira/browse/FLINK-31867.

Reply:
Maybe a better solution would be to throw an exception and clear the autoscaler info completely. That will basically reset the history.